lib/parsers/m3u_parser.rb

```diff
@@ -0,0 +1,22 @@
+class FormatParser::M3UParser
+  include FormatParser::IOUtils
+
+  HEADER = '#EXTM3U'
+
+  def likely_match?(filename)
+    filename =~ /\.(m3u|m3u8)$/i
```

Comment:
```suggestion
    filename =~ /\.m3u8?$/i
```

This should shorten things a bit.